command/loadbot/generator/contract_txn_generator.go

```diff
@@ -0,0 +1,71 @@
+package generator
+
+import (
+	"math/big"
+	"sync/atomic"
+
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+type ContractTxnsGenerator struct {
+	BaseGenerator
+
+	contractBytecode []byte
+	encodedParams    []byte
+}
+
+//	Returns contract deployment tx if contractAddress is empty, otherwise returns
+//	a token transfer tx
+func (gen *ContractTxnsGenerator) GetExampleTransaction() (*types.Transaction, error) {
+	if gen.contractAddress == nil {
+		//	contract not deployed yet
+		//	generate contract deployment tx
+		return gen.signer.SignTx(&types.Transaction{
+			From:     gen.params.SenderAddress,
+			Value:    big.NewInt(0),
+			GasPrice: gen.params.GasPrice,
+			Input:    gen.contractBytecode,
+			V:        big.NewInt(1), // it is necessary to encode in rlp
+		}, gen.params.SenderKey)
+	}
+
+	//	return token transfer tx
+	return gen.signer.SignTx(&types.Transaction{
+		From:     gen.params.SenderAddress,
+		To:       gen.contractAddress,
+		Value:    big.NewInt(0),
+		GasPrice: gen.params.GasPrice,
+		Input:    gen.encodedParams,
+		V:        big.NewInt(1), // it is necessary to encode in rlp
+	}, gen.params.SenderKey)
+}
```

Comment:
I don't think we need if case here because nil is set to To in if block.

Reply:
That method checks if the Contract Address is set.
If it is not it sets the Contract Bytecode in the Input field, and if it is, sets the Encoded Parameters which are needed to call SC function.

So anyway, we need to set some kind of flow check in order to set different Input field.
We could basically check if To field is nul, but the if statement would remain.
Unless you have a better idea ? Please do share 🙂

Reply:
OK. So Input is a different field besides To